inc/c_util/tarray_ll.h

```diff
@@ -0,0 +1,183 @@
+                    /*Codes_SRS_TARRAY_02_009: [ If there are any failures then TARRAY_ENSURE_CAPACITY(T) shall fail and return a non-zero value. ]*/           \
+                    LogError("failure in realloc(tarray->arr=%p, new_capacity=%" PRIu32 " * sizeof(T)=%zu)",                                                    \
+                        tarray->arr, new_capacity, sizeof(T));                                                                                                  \
+                    result = MU_FAILURE;                                                                                                                        \
+                }                                                                                                                                               \
+                else                                                                                                                                            \
+                {                                                                                                                                               \
+                    /*Codes_SRS_TARRAY_02_008: [ TARRAY_ENSURE_CAPACITY(T) shall succeed, record the new computed capacity, and return 0. ]*/                   \
+                    ((TARRAY_TYPEDEF_NAME(T)*)tarray)->capacity = new_capacity;                                                                                 \
+                    ((TARRAY_TYPEDEF_NAME(T)*)tarray)->arr = temp;                                                                                              \
+                    result = 0;                                                                                                                                 \
+                }                                                                                                                                               \
+            }                                                                                                                                                   \
+        }                                                                                                                                                       \
+    }                                                                                                                                                           \
+    return result;                                                                                                                                              \
+}
+
+/*macro to be used in headers*/                                                                                     \
+#define TARRAY_LL_TYPE_DECLARE(C, T)                                                                                \
+    /*hint: have TARRAY_DEFINE_STRUCT_TYPE(T) before TARRAY_LL_TYPE_DECLARE*/                                       \
```

Comment:
>TARRAY_DEFINE_STRUCT_TYPE [](http://example.com/codeflow?start=17&length=25)

This comment makes me wonder what I'm supposed to do when defining my own TARRAY and whether I need to use the LL outside of tests...but I guess I should look at some examples in the tests first.

Reply:
Although it seems like we should have an example in the comments here or (probably better) have an MD at least with the "how to use this" example. I see some SRS in this file but no MD in the PR

Reply:
Added a comment near TARRAY that tarray_int tests can be used a template for tests/reals.